MediumProject/src/main/java/stage4/SearchingEngine.java

```diff
@@ -0,0 +1,29 @@
+package stage4;
+import java.util.List;
+public class SearchingEngine {
+    private static final List<Person> generalListOfUsers = List.copyOf(UserDetails.getPeople());
+
+    public static void searchPerson(String searchingWord){
+        List<Person> foundUsers = searchUsers( searchingWord);
+        printSearchResult(foundUsers);
+    }
+
+    private static List<Person> searchUsers( String searchingWord) {
+        return generalListOfUsers.stream().findAny()
+                .stream()
+                .filter(p -> p.getFirstName().toLowerCase().contains(searchingWord) ||
```

Comment:
will it work if you add toString() or custom getPersonDetails() method which will concat all 3 attributes

                .filter(p -> p.getFirstName().toLowerCase().contains(searchingWord) ||
                        p.getLastName().toLowerCase().contains(searchingWord) ||
                        p.getEmail().toLowerCase().contains(searchingWord))

=>>>>>

                 .filter(p -> p.getPersonDetails().toLowerCase().contains(searchingWord) )

in Person class 

 public String getPersonDetails(){
return this.firstName+this.lastName+.....
}

Reply:
btw you can use toString() -


    @Override
    public String toString() {
        if (email.isBlank()) return firstName + " " + lastName;
        else return firstName + " " + lastName + " " + email;
    }

Reply:
done